tensorflow_lite_support/c/task/vision/image_classifier.cc

```diff
@@ -0,0 +1,223 @@
+}
+
+TfLiteClassificationResult* GetClassificationResultCStruct(
+    const ClassificationResultCpp classification_result_cpp) {
+  auto* c_classifications =
+      new TfLiteClassifications[classification_result_cpp
+                                    .classifications_size()];
+
+  for (int head = 0; head < classification_result_cpp.classifications_size();
+       ++head) {
+    const ClassificationsCpp& classifications =
+        classification_result_cpp.classifications(head);
+    c_classifications[head].head_index = head;
+
+    auto* c_categories = new TfLiteCategory[classifications.classes_size()];
+    c_classifications->size = classifications.classes_size();
+
+    for (int rank = 0; rank < classifications.classes_size(); ++rank) {
+      const ClassCpp& classification = classifications.classes(rank);
+      c_categories[rank].index = classification.index();
+      c_categories[rank].score = classification.score();
+
+      if (classification.has_class_name())
+        c_categories[rank].label = strdup(classification.class_name().c_str());
+      else
+        c_categories[rank].label = nullptr;
+
+      if (classification.has_display_name())
+        c_categories[rank].display_name =
+            strdup(classification.display_name().c_str());
+      else
+        c_categories[rank].display_name = nullptr;
+    }
+    c_classifications[head].categories = c_categories;
+  }
+
+  TfLiteClassificationResult* c_classification_result =
+      new TfLiteClassificationResult;
+  c_classification_result->classifications = c_classifications;
+  c_classification_result->size =
+      classification_result_cpp.classifications_size();
+
+  return c_classification_result;
+}
+
+TfLiteClassificationResult* TfLiteImageClassifierClassify(
+    const TfLiteImageClassifier* classifier,
+    const TfLiteFrameBuffer* frame_buffer) {
+  std::unique_ptr<FrameBufferCpp> cpp_frame_buffer =
+      CreateCppFrameBuffer(frame_buffer);
+
+  if (cpp_frame_buffer == nullptr) return nullptr;
+
+  StatusOr<ClassificationResultCpp> classification_result_cpp =
+      classifier->impl->Classify(*cpp_frame_buffer);
+
+  if (!classification_result_cpp.ok()) return nullptr;
+
+  TfLiteClassificationResult* c_classification_result =
+      GetClassificationResultCStruct(
+          ClassificationResultCpp(classification_result_cpp.value()));
+
+  return c_classification_result;
+}
+
+TfLiteClassificationResult* TfLiteImageClassifierClassifyWithRoi(
+    const TfLiteImageClassifier* classifier,
+    const TfLiteFrameBuffer* frame_buffer, const TfLiteBoundingBox* roi) {
+  BoundingBoxCpp cc_roi;
+  cc_roi.set_origin_x(roi->origin_x);
+  cc_roi.set_origin_y(roi->origin_y);
+  cc_roi.set_width(roi->width);
+  cc_roi.set_height(roi->height);
+
+  std::unique_ptr<FrameBufferCpp> cpp_frame_buffer =
+      CreateCppFrameBuffer(frame_buffer);
+  StatusOr<ClassificationResultCpp> classification_result_cpp =
+      classifier->impl->Classify(*cpp_frame_buffer, cc_roi);
```

Comment:
What if `cpp_frame_buffer` is null?

Reply:
Fixed. Now CreateCppFrameBuffer returns StatusOr